src/api/types.ts

```diff
@@ -251,7 +251,7 @@ export interface DataSource<T> {
   /**
    * Fetches arbitrary data from a data source
    */
-  fetchData(searchTerm?: string): Promise<Array<T>>;
+  fetchData(searchTerm?: string, config?: Record<string, any>): Promise<Array<T>>;
```

Comment:
Should this be `config` or should it be `parameters` or something since it isn't really "configuring"; it's providing additional arguments.